core/modules/ccontrol/UserQueryFactory.cc

```diff
@@ -186,6 +176,16 @@ UserQueryFactory::newUserQuery(std::string const& aQuery,
         LOGS(_log, LOG_LVL_DEBUG, "Old-style Hierarchy: " << *a2stmt);
 #endif
 
+        // parse using antlr4
+        std::shared_ptr<query::SelectStmt> stmt;
+        try {
+            stmt = a4NewUserQuery(query);
+        } catch (parser::adapter_order_error& e) {
+            return std::make_shared<UserQueryInvalid>(std::string("ParseException:") + e.what());
+        } catch (parser::adapter_execution_error& e) {
+            return std::make_shared<UserQueryInvalid>(std::string("ParseException:") + e.what());
+        }
+
```

Comment:
there's no change here, except that the block that generates antlr4 code got moved down.